presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -708,7 +710,19 @@ protected Type visitFunctionCall(FunctionCall node, AnalysisContext context)
                 argumentTypes.add(process(expression, context).getTypeSignature());
             }
 
-            Signature function = functionRegistry.resolveFunction(node.getName(), argumentTypes.build(), context.isApproximate());
+            Signature function;
+            try {
+                function = functionRegistry.resolveFunction(node.getName(), argumentTypes.build(), context.isApproximate());
+            }
+            catch (PrestoException e) {
```

Comment:
At some point we should change the function registry to return an Optional and let the analyzer throw the appropriate exception.